train.py

```diff
@@ -381,7 +389,7 @@ def loss_fn(pred, labels):
                 # pipeline parallel forward / backward inside step() call
                 is_last_stage = pp_mesh.get_local_rank() == pp_mesh.size() - 1
 
-                with train_context():
+                with train_context() as train_contexts:
```

Comment:
nit: should either rename `train_context()` to `train_contexts()` or `as train_contexts` to `as train_context`

Reply:
I don't think you can because that would be assigning a dictionary and a function to train_context then

Reply:
oof good point haha